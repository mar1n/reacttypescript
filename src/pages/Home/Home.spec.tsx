import React from "react";
import renderer from "react-test-renderer";

import HomePage, { filteredRepos } from "./Home.page";

test("Homepage renders as expected", () => {
  const component = renderer.create(
    <HomePage/>
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});


test('filterRepos() will reduce the repo array based on passed search term', () => {
  const dummyRepos = [
    {
      id: 1,
      name: 'This is a name'
    },
    {
      id: 2,
      name: 'another name'
    },
    {
      id: 3,
      name: 'This is a name'
    },
    {
      id: 4,
      name: 'another name'
    }
  ]
 
  const expectedResult: any = [
    {
      id: 1,
      name: 'This is a name'
    },
    {
      id: 3,
      name: 'This is a name'
    }
  ]

  const actualResult = filteredRepos(dummyRepos, 'is')

  expect(actualResult).toEqual(expectedResult)
})