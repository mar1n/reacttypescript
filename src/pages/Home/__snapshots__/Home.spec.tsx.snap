// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders as expected 1`] = `
<div
  className="App"
>
  <main
    className="sc-AxhCb edBHIa"
  >
    <div>
      <input
        className="sc-AxjAm kXckSO"
        onChange={[Function]}
        value=""
      />
    </div>
    <div
      className="sc-AxirZ bNRZuW"
    >
      Loading...
    </div>
  </main>
</div>
`;
